File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func removeNthFromEnd(head *ListNode, n int) *ListNode {
    cur := head
    count := 1
    
    for cur.Next != nil {
        count++
        cur = cur.Next
    }
    
    if count - n == 0 {
        return head.Next
    }
    
    stitchIndex := count - n
    cur = head
    for i := 1; i < count + 1; i++ {
        if i == stitchIndex {
            tmp := cur.Next
            next := tmp.Next
            cur.Next = next
            return head
        }
        
        cur = cur.Next
    }
    
    return nil
    
}

File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        if head.next == None:
            return None
        
        cur = head
        count = 1
        
        while cur.next != None:
            count += 1
            cur = cur.next
        
        prev = head
        cur = head.next
        target = count - n + 1
        
        if target == 1:
            return head.next
        
        for i in range(2, count+1):
            if i == target:
                prev.next = cur.next
                return head
                
            prev = prev.next    
            cur = cur.next
        
        return head
                
                
            
            
        
        
        